From a59694101d610009d5d75b209b9d290b64039c44 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:47:55 -0400
Subject: [powerpc] pci: Use list_for_each_entry() for bus traversal

Message-id: <20140806034755.1438.15795.stgit@gir.stowe>
Patchwork-id: 87445
O-Subject: [RHEL7.1 PATCH 36/69] powerpc/PCI: Use list_for_each_entry() for bus traversal
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 3bc955987fb377f3c95bc29deb498e96819b8451

commit 3bc955987fb377f3c95bc29deb498e96819b8451
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Feb 13 21:14:01 2014 +0800

    powerpc/PCI: Use list_for_each_entry() for bus traversal

    Replace list_for_each() + pci_bus_b() with list_for_each_entry().

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/pci_64.c b/arch/powerpc/kernel/pci_64.c
index cdf5aa1..97cc4bb 100644
--- a/arch/powerpc/kernel/pci_64.c
+++ b/arch/powerpc/kernel/pci_64.c
@@ -208,7 +208,6 @@ long sys_pciconfig_iobase(long which, unsigned long in_bus,
      unsigned long in_devfn)
 {
  struct pci_controller* hose;
- struct list_head *ln;
  struct pci_bus *bus = NULL;
  struct device_node *hose_node;
 
@@ -230,8 +229,7 @@ long sys_pciconfig_iobase(long which, unsigned long in_bus,
   * used on pre-domains setup. We return the first match
   */
 
- for (ln = pci_root_buses.next; ln != &pci_root_buses; ln = ln->next) {
-  bus = pci_bus_b(ln);
+ list_for_each_entry(bus, &pci_root_buses, node) {
   if (in_bus >= bus->number && in_bus <= bus->busn_res.end)
    break;
   bus = NULL;
diff --git a/arch/powerpc/platforms/pseries/pci_dlpar.c b/arch/powerpc/platforms/pseries/pci_dlpar.c
index efe6137..203cbf0 100644
--- a/arch/powerpc/platforms/pseries/pci_dlpar.c
+++ b/arch/powerpc/platforms/pseries/pci_dlpar.c
@@ -37,15 +37,15 @@ find_bus_among_children(struct pci_bus *bus,
                         struct device_node *dn)
 {
  struct pci_bus *child = NULL;
- struct list_head *tmp;
+ struct pci_bus *tmp;
  struct device_node *busdn;
 
  busdn = pci_bus_to_OF_node(bus);
  if (busdn == dn)
   return bus;
 
- list_for_each(tmp, &bus->children) {
-  child = find_bus_among_children(pci_bus_b(tmp), dn);
+ list_for_each_entry(tmp, &bus->children, node) {
+  child = find_bus_among_children(tmp, dn);
   if (child)
    break;
  };
-- 
1.7.1